### ISSUE TITLE
Tool crashes when run on arbitrary folder

### ISSUE THREAD

**Issue by Inventitech** [ClintonCao/UnifiedASATVisualizer#130]
Currently, the tool crashes when I run it on, for example, the top-level repo folder of this repository (even though I ran `maven site` before) Of course, I did not add ASATs to its pom.xml. There is no output, and it does not fail gracefully. It is also unclear to me, whether one should point it to the target folder or the top-level dir.
1. Make tool not deadlock on adding a folder that is not suitable.
2. Add short documentation on which folder to select.

**Comment by ClintonCao** [comment]
So when you started up the jar and then selected `ContextProject-TSE`, the tool crashes? 
There is no need to add the ASATs because there are already in the pom.xml; running maven site should have created the output files for the ASATs. I assume that you are running the jar from the terminal? If so, did you get a stack trace in the terminal?.

And regarding which folder to select, in the README, there are instructions on how to run the program. The last step was selecting the project folder (this is also in the GUI when the program is started). But I guess selecting project folder might be a bit confusing?

**Comment by Inventitech** [comment]
Yeah, the tool consumes 100% CPU power and freezes (killed it after ~10 minutes). Absolutely no output :-)
I just checked, and indeed, you have the ASATs configured for the project. So it crashes even on non-arbitrary folders ;-)

> And regarding which folder to select, in the README, there are instructions on how to run the program. The last step was selecting the project folder (this is also in the GUI when the program is started). But I guess selecting project folder might be a bit confusing?

Selecting the project is ambiguous. I tried to change this in #131.

**Comment by ClintonCao** [comment]
Hmm...Just to make sure, do you have Chrome installed? The program also stalls on my Ubuntu VM because I don't have Chrome installed.

**Comment by Inventitech** [comment]
Yeah, I have Chrome.

I just debugged into this issue. 

This is the stacktrace:



The surfacing problem is that you don't catch the NoSuchElementException exception that get in

```Java
// for-loop in stream.
				String filePath = ProjectInfoFinder.getClassPaths().stream().filter(p -> p.endsWith(fileNWithSep)).findFirst().get();
``` 

can throw.

However, even after commenting 
```
		parseFile(new PMDXMLParser(), JavaController.getPmdOutputFile());
``` 

in JSONFormatter.java, it still hangs (somewhere else, I don't know where). Good news: I just found out that it never "frees" the UI, but the json is actually being written just fine. 

![image](https://cloud.githubusercontent.com/assets/334256/19814723/cfe61790-9d3f-11e6-88d6-7ac13dd36b3c.png)

Example of "successful" run of Java application.
